e2e_tests/tests/cluster/test_agent_restart.py

```diff
@@ -95,24 +100,25 @@ def test_agent_restart_exp_container_failure(
         restartable_managed_cluster.restart_agent()
         # As soon as the agent is back, the original allocation should be considered dead,
         # but the new one should be allocated.
-        state = exp.experiment_state(sess, exp_id)
+        exp_ref.reload()
         # old STATE_ACTIVE got divided into three states
-        assert state in [
-            bindings.experimentv1State.ACTIVE,
-            bindings.experimentv1State.QUEUED,
-            bindings.experimentv1State.PULLING,
-            bindings.experimentv1State.STARTING,
-            bindings.experimentv1State.RUNNING,
+        assert exp_ref.state in [
+            client.ExperimentState.ACTIVE,
+            client.ExperimentState.QUEUED,
+            client.ExperimentState.PULLING,
+            client.ExperimentState.STARTING,
+            client.ExperimentState.RUNNING,
         ]
-        exp.wait_for_experiment_state(sess, exp_id, bindings.experimentv1State.RUNNING)
+        exp.wait_for_experiment_state(sess, exp_ref.id, bindings.experimentv1State.RUNNING)
         tasks_data = _task_list_json(sess)
         assert len(tasks_data) == 1
         exp_task_after = list(tasks_data.values())[0]
 
         assert exp_task_before["taskId"] == exp_task_after["taskId"]
         assert exp_task_before["allocationId"] != exp_task_after["allocationId"]
 
-        exp.wait_for_experiment_state(sess, exp_id, bindings.experimentv1State.COMPLETED)
+        exp_ref.kill()
+        assert exp_ref.wait(interval=0.01) == client.ExperimentState.CANCELED
```

Comment:
Nice! We don't have to wait for the exp completion, which means we can save time!

Reply:
yeah there were several of those that I found.